### PR TITLE
Add ticks in case there are not enough in the SymmetricalLogLocator in Ticker.py

### PR DESCRIPTION
**Pull request by Schmutsi**
## PR Summary

In some cases, when a graph was ploted with the symLog scale, we didn't have enough ticks on the y axe. Now, new ticks are added when there are not enough to undersand the graph.
We just had a condition at the end of the function tick_values() in the class SymetricalLogLocator().

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by jklymak** [comment]
Thanks for the PR!.  Can you fix the title to not have a back reference (this can go in the description as "Closes #12345") and include a description of the PR in the description?  Further the docs need to build.  If they don't that is a sign something broke.  This probably also needs new tests to make sure your desired behaviour is correct.

**Comment by Schmutsi** [comment]
Excuse me but I don't understand what you mean when you say "the docs need to build". And I don't understand there are flake8 errors while I didn't check the case in the PR checklist.

**Comment by jklymak** [comment]
We can't merge anything until all the tests pass.  See the red x-s where it says "Some checks were not successful"

**Comment by QuLogic** [comment]
Why did you remove every single blank line in every docstring? You'll need to at least revert that to get docs to work.

**Comment by Schmutsi** [comment]
Thanks you for the tips for every single blank line in every docstring. This issue is fixed.
Yet, I still don't understand my two last failures.
     - "matplotlib.matplotlib Failing after 31m — Build #20201207.2 had test failures"
     - "matplotlib.matplotlib (Job Windows_py37) Failing after 13m — Job Windows_py37 failed"
Could you explain me what is the problem please ?

**Comment by dopplershift** [comment]
The test failing is: `lib/matplotlib/tests/test_backend_tk.py::test_figuremanager_preserves_host_mainloop` That seems unrelated to this PR?

**Comment by jklymak** [comment]
So now this needs a test to prove that you fixed what you claim you fixed and to make sure that it does not break in the future....

**Comment by Schmutsi** [comment]
Our solution is working but we found an issue that doesn't concern directly our issue. The missing ticks are displayed for the symloglocator, but sometimes the formatter doesn't work well. Indeed, as you see in this picture, the ticks are here but not their labels.

original_formatter

The issue comes from the LogFormatter : we had to change the minor_thresholds in order to display all the labels. We don't know if it is the right way to fix the issue but it works for our case as you can see in the picture below.

our_formatter

We changed the values of the minor_thresholds from (1, 0.4) to (3, 3)

formatter_modified

**Comment by Schmutsi** [comment]
Come back to our pull request.
It seems that there is an issue with the "Tests / Python 3.7 on ubuntu-16.04 (Minimum Versions) (pull_request)" that block the merging but I don't understand what is probleme (we just changed the test_ticker.py and it doesn't seem affect by the error).
Could I have an explanation please ?

**Comment by QuLogic** [comment]
It is unrelated; I restarted the jobs.

File: lib/matplotlib/tests/test_ticker.py [hunk 1]
```diff
@@ -443,6 +443,22 @@ def test_set_params(self):
...
+    def test_min_ticks(self):
+        """
+        Create 6 graphs and test if all of theses graphs have 3 ticks minimum
+        on the y axis. Values have been choosen in order to cover all the
+        cases. With only positive/negative value, both positive and negative
+        values, with close values an with distant values.
+        """
+        st = [-2.5, -2, -1, -0.01, 2, 2.5]
+        en = [-200, -2.005, 1, 0.01, 2.005, 200]
+        fig, axen = plt.subplots(2, 3)
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_ticker.py:455]:
```suggestion
        starts = [-2.5, -2, -1, -0.01, 2, 2.5]
        ends = [-200, -2.005, 1, 0.01, 2.005, 200]
        fig, axs = plt.subplots(2, 3)
```

`starts` and `ends` is more readable, and `axs` is the canonical name we use for the Axes'es returned by `subplots()`.

File: lib/matplotlib/ticker.py [hunk 2]
```diff
@@ -2698,6 +2698,10 @@ def get_log_range(lo, hi):
                     ticklocs.extend(subs * decade)
         else:
             ticklocs = decades
+            # if there is not enough ticks to show on the graph
+            if len(ticklocs) <= 2:
+                subs = np.linspace(vmin, vmax, 5)
```
Comment (timhoffm) [inline lib/matplotlib/ticker.py:2703]: Two questions:

1. Why is this 5? Shouldn't 3 be sufficient (or even 3 - ticklocs)?
2. Can you convince me that this is the right fix? I'm not quite into the locator code, but from a quick look I have the impression that if there are less than two decades, there should be something similar to the above `subs = np.arange(2.0, base)` handling, i.e. place labels on all the minor ticks (or maybe even on a subset of them) - but not on "arbitrary" locations.

**Comment by jklymak** [comment]
I'm going to close this, but anyone should feel free to take it up...

**Pull request closed without merging**